packages/presentation/lib/app/app_bloc.dart

```diff
@@ -108,7 +118,14 @@ class _AppBloc extends BlocImpl implements AppBloc {
   @override
   void onSelectedTab(int index) {
     if (selectedTab == index) return;
-    selectedTab = index;
+    switch (index) {
+      case 0:
+        _popAllAndPush(bottomNavBarStack[index]!.call());
+        break;
+      case 3:
+        _popAllAndPush(bottomNavBarStack[index]!.call());
+        break;
+    }
```

Comment:
do it via enum. so, u will have the following flow:

1) u will have a enum for bottom navigation

enum BottomNavigationItemType {
    home
    ...
    ...
    login
}

2) u can create a static method/extension, which consumes number as parameter and returns enum item (as each enum has its own index

3) u refactor your code like this

final type = BottomNavigationItemType.fromIndex(index)
switch (type) {
    case BottomNavigationItemType.home -> do smth

it seems much prettier

Reply:
realized by trying new enum.  but in fact, you can leave one line of the transition code, since our index is always equal to